packages/primitives/src/Label/Label.tsx

```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import { Box } from '../Box'
+
+type LabelProps = JSX.IntrinsicElements['label'] &
+  React.ComponentProps<typeof Box>
+
+export const Label = React.forwardRef<HTMLLabelElement, LabelProps>(
+  ({ variant = 'label', ...props }, ref) => (
+    <Box as="label" variant={variant} ref={ref as any} {...props} />
```

Comment:
@alexanderchan I just checked and we still have the problem of `as` not being properly typed when wrapping `Box`. For example `htmlFor` does not get typed properly with this `Label` component. We are missing both of these because we are still typing the Label component as a `div`. https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L2058

I am still leaning towards not creating all components as wrappers for Box. I would rather ensure that we have a better typed system than a more concise API at this points since one of the benefits of Kodiak over Theme UI is that it is built on TS. 

I have the same concern with losing the typing of the forwarded ref. 

What do you think?

Reply:
@wesleycole ok 👍  guess we can compose from the styled functions so that we can keep the refs working until we can work through the typing.

Reply:
Yea, I think if this was truly an internal project or only for Jilt, I would be fine with this approach but I want to make sure that we are typing everything as well as possible and not introducing any unexpected behavior since this could potentially be used in any project.